autogen/browser_utils/browser_creator.py

```diff
@@ -0,0 +1,19 @@
+from bing_browser import BingTextBrowser
+from google_browser import GoogleTextBrowser
+
+
+class TextBrowserCreator:
+    """Creator class for creating different text browsers at runtime. Make sure to add newly registered browsers here"""
+    
+    browser_classes = {
+        'BingTextBrowser': BingTextBrowser,
+        'GoogleTextBrowser': GoogleTextBrowser
+    }
+
+    @classmethod
+    def create_browser(cls, name: str):
+        """Factory method to create a text browser instance based on the name."""
+        if name in cls.browser_classes:
+            return cls.browser_classes[name]()
```

Comment:
if we need doing this look up we could really just use enums : https://docs.python.org/3/library/enum.html